pkg/migrator/migrator.go

```diff
@@ -40,39 +40,39 @@ func New(client client.Client) Migrator {
 }
 func (m *migrator) Migrate() error {
 
-	cscs := &v2.CatalogSourceConfigList{}
-	err := m.client.List(context.TODO(), &client.ListOptions{}, cscs)
+	opsrcs := &v1.OperatorSourceList{}
```

Comment:
Given that the OpSrc CRD was removed from the manifests dir, shouldn't CVO delete the OpSrc CRD from the cluster? If CVO removes the CRD, these CatSrc's will be removed, which shouldn't be a problem if Marketplace recreates them on startup.

Reply:
The migrator is more for any non-default operatorsources that weren't cleared out by the admin. 
CVO will block the upgrade if there are non-default operatorsources in the cluster before upgrade, but the migrator's job is to clean up the resources if that block from CVO was overridden somehow. 

All the migrator does, is simply remove the owner refs from the child resources(i.e CatalogSource in this case) and deleted the operatorsource.

Reply:
Am I missing something? This code will only be available when 4.6 is released.In 4.6, openshift will no longer have the OpSrc and CatSrc CRDs from the cluster. How will marketplace remove a CR for a CRD that doesn't exist?

Reply:
Deleting a CRD when a CR has a finalizer will prevent the CRD from being deleted as outlined in this [issue](https://github.com/kubernetes/kubernetes/issues/60538).